tests/dashbio_demos/app_forna_container.py

```diff
@@ -0,0 +1,585 @@
+                                        dcc.Input(
+                                            id='forna-color-domain-high',
+                                            type='number',
+                                            value=100
+                                        ),
+                                        html.Br(),
+                                        html.Br(),
+                                        html.Div(
+                                            className='fullwidth-app-controls-name',
+                                            children='Colors map'
+                                        ),
+                                        html.Div(
+                                            className='app-controls-desc',
+                                            children='Specify the colors for each ' +
+                                            'nucleotide by entering the position of ' +
+                                            'the nucleotide into the left input box, ' +
+                                            'and either a) a string representation ' +
+                                            'of a color or b) a number within the ' +
+                                            'range specified above. Then, press the ' +
+                                            '"Submit" button,'
+                                        ),
+                                        html.Br(),
+                                        dcc.Input(
+                                            id='forna-color-map-nucleotide',
+                                            type='number',
+                                            min=1,
+                                            placeholder=1
+                                        ),
+                                        dcc.Input(
+                                            id='forna-color-map-color',
+                                            placeholder='green'
+                                        ),
+                                        html.Br(),
+                                        html.Br(),
+                                        html.Button(
+                                            id='forna-submit-custom-colors',
+                                            children='Submit'
+                                        )
+                                    ]
+                                )
+                            ])
+                        )
+                    ])
+                ]),
+            html.Div(id='forna-container', children=[
+                dash_bio.FornaContainer(
+                    id='forna',
+                    height=500,
+                    width=500
+                )
+            ]),
+
+            dcc.Store(id='forna-sequences', data=initial_sequences),
+            dcc.Store(id='forna-custom-colors')
+        ]
+    )
+
+
+def callbacks(app):  # pylint: disable=redefined-outer-name
```

Comment:
is this `pylint` directive necessary?

Reply:
Yes, it's in all of the apps. It's because we have the option to either run them as standalone or as "sub-apps" of the app defined in `index.py`: 

https://github.com/plotly/dash-bio/blob/729402c5e20dcd0425bf5bbe52dedad99a93721f/tests/dashbio_demos/app_clustergram.py#L901
https://github.com/plotly/dash-bio/blob/729402c5e20dcd0425bf5bbe52dedad99a93721f/index.py#L14

The `callbacks` function needs to be defined in each app. To attach the callbacks to the "super" app, the function takes a Dash app as an argument and is called whenever `index.py` is run. 

I'm guessing that naming the argument to the `callbacks` function `app` is resulting in this problem. I can rename it.

Reply:
ah of course, `app` is redefined down below - good catch. Yes, would be better to make the name unique inside this function.